pubsub/iterator.go

```diff
@@ -352,12 +358,29 @@ func (it *messageIterator) recvMessages() ([]*pb.ReceivedMessage, error) {
 	if err != nil {
 		return nil, err
 	}
-	it.eoMu.Lock()
-	if got := res.GetSubscriptionProperties().GetExactlyOnceDeliveryEnabled(); got != it.enableExactlyOnceDelivery {
+
+	// If the new exactly once settings are different than the current settings, update it.
+	it.eoMu.RLock()
+	enableEOD := it.enableExactlyOnceDelivery
+	it.eoMu.RUnlock()
+
+	if got := res.GetSubscriptionProperties().GetExactlyOnceDeliveryEnabled(); got != enableEOD {
```

Comment:
nit: create a variable `subprop := res.GetSubscriptionProperties()` to reused on both check.